framework/include/multiapps/MultiApp.h

```diff
@@ -40,6 +40,58 @@ class NumericVector;
 template <>
 InputParameters validParams<MultiApp>();
 
+/// Holds app partitioning information relevant to the a particular rank for a
+/// multiapp scenario.
+struct LocalRankConfig
+{
+  /// The number of apps that should/will be run locally on this rank.
+  unsigned int num_local_apps;
+  /// The (global) index of the first local app for this rank.  All ranks that
+  /// are used to perform multi-proc parallel runs for a given app will have the
+  /// same first_local_app_index as each other.
+  unsigned int first_local_app_index;
+  /// This is true if this rank is the primary/zero rank for a (sub)app slot.
+  /// A slot is all ranks that are grouped together to run a single (sub)app
+  /// together.  This field will be true for exactly one rank in each slot.
+  /// This is important for things like multiapp transfers where you want to
+  /// only transfer data to a given subapp once even though it may be running on
+  /// multiple procs/ranks.
+  bool is_first_local_rank;
+};
+
+inline bool
+operator==(const LocalRankConfig & lhs, const LocalRankConfig & rhs)
+{
+  return (lhs.num_local_apps == rhs.num_local_apps) &&
+         (lhs.first_local_app_index == rhs.first_local_app_index) &&
+         (lhs.is_first_local_rank == rhs.is_first_local_rank);
+}
+
+inline bool
+operator!=(const LocalRankConfig & lhs, const LocalRankConfig & rhs)
+{
+  return !(lhs == rhs);
+}
+
+/// Returns app partitioning information relevant to the given rank for a
+/// multiapp scenario with the given number of apps (napps) and parallel/mpi
+/// procs (nprocs).  min_app_procs and max_app_procs define the min and max
+/// number of procs that must/can be used in parallel to run a given (sub)app.
+///
+/// Each proc calls this function in order to determine which (sub)apps among
+/// the global list of all subapps for a multiapp should be run by the given
+/// rank.
+LocalRankConfig rankConfig(unsigned int rank,
+                           unsigned int nprocs,
+                           unsigned int napps,
+                           unsigned int min_app_procs,
+                           unsigned int max_app_procs);
+
+unsigned int nSlots(unsigned int nprocs,
+                    unsigned int napps,
+                    unsigned int min_app_procs,
+                    unsigned int max_app_procs);
+
```

Comment:
Should this be moved to MooseUtils.C/h? It is used by multiple objects and will probably be used by others in the future.

Reply:
This function is gone now.